File: wallet-kit/src/main/java/bitcoin/walllet/kit/network/message/GetAddrMessage.java
package bitcoin.walllet.kit.network.message;

import java.io.IOException;

/**
 * Build P2P message:
 * https://en.bitcoin.it/wiki/Protocol_documentation#Message_structure
 *
 * @author liaoxuefeng
 */
public class GetAddrMessage extends Message {

    public GetAddrMessage() {
        super("getaddr");
    }

    public GetAddrMessage(byte[] payload) throws IOException {
        super("getaddr");
    }

    @Override
    protected byte[] getPayload() {
        return new byte[0];
    }

    @Override
    public String toString() {
        return "GetAddrMessage()";
    }

}

File: wallet-kit/src/main/kotlin/bitcoin/wallet/kit/messages/GetAddrMessage.kt
package bitcoin.wallet.kit.messages

import java.io.IOException

/**
 * The 'getaddr' message is sent to a peer to request a list of known peers.
 * The 'addr' message is a response
 */
class GetAddrMessage : Message {

    constructor() : super("getaddr")

    @Throws(IOException::class)
    constructor(payload: ByteArray) : super("getaddr")

    override fun getPayload(): ByteArray {
        return ByteArray(0)
    }

    override fun toString(): String {
        return "GetAddrMessage()"
    }
}
